velox/connectors/hive/HiveConnector.h

```diff
@@ -345,4 +345,36 @@ class HiveHadoop2ConnectorFactory : public HiveConnectorFactory {
       : HiveConnectorFactory(kHiveHadoop2ConnectorName) {}
 };
 
+class HivePartitionFunctionSpec : public core::PartitionFunctionSpec {
+ public:
+  HivePartitionFunctionSpec(
+      int numBuckets,
+      const std::vector<int> bucketToPartition,
```

Comment:
I'm not sure about this one, this will enforce the caller to pass in rvalue, but that's not necessary; Also I think it would be good if we keep consistent with other Spec definition

Reply:
That's ok. Then we you should pass by value as you move these input objects in ctor. If you pass const reference, then don't move those inputs in ctor.